forem/forem
# Add ability to bulk search tags by name or id

## What type of PR is this? (check all applicable)


## Description

Created a "bulk" tags controller action with the ability to search for tags by name or id in bulk. There was some back in forth as to where the endpoint should be updated with the new filtering. Ultimately landed on creating a new controller action (`bulk`) for internal use. This meant ripping out what was in the V0 api in favor of this approach. This is why you'll see a lot of deletions in this PR.

## Related Tickets & Documents

- Related Reverted Merge: https://github.com/forem/forem/commit/c76909a428b23e5941c60f6d9561db22aaa618da

## QA Instructions, Screenshots, Recordings

Visit `/tags/bulktag_ids[]=2&tag_ids[]=10` or `/tags/bulk?tag_names[]=ruby&tag_names[]=javascript` and you should see different payload based on what was searched.

### UI accessibility concerns?

None as it's just an update to api endpoint response.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
